gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
+      assertNull(remoteDisk.id());
+      assertTrue(remoteDisk.configuration() instanceof StandardDiskConfiguration);
+      assertNull(remoteDisk.configuration().sizeGb());
+      assertEquals("pd-ssd", remoteDisk.configuration().diskType().diskType());
+      assertEquals(DiskConfiguration.Type.STANDARD, remoteDisk.configuration().type());
+      assertNull(remoteDisk.lastAttachTimestamp());
+      assertNull(remoteDisk.lastDetachTimestamp());
+      count++;
+    }
+    assertEquals(2, count);
+    // test snapshots
+    SnapshotId firstSnapshotId = SnapshotId.of(diskNames[0]);
+    SnapshotId secondSnapshotId = SnapshotId.of(diskNames[1]);
+    firstOperation = compute.create(SnapshotInfo.of(firstSnapshotId, firstDiskId));
+    secondOperation = compute.create(SnapshotInfo.of(secondSnapshotId, secondDiskId));
+    while (!firstOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    while (!secondOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    // test list snapshots
+    Compute.SnapshotFilter snapshotFilter =
+        Compute.SnapshotFilter.equals(Compute.SnapshotField.NAME, prefix + "\\d");
+    Page<Snapshot> snapshotPage =
+        compute.listSnapshots(Compute.SnapshotListOption.filter(snapshotFilter));
+    Iterator<Snapshot> snapshotIterator = snapshotPage.iterateAll();
+    count = 0;
+    while (snapshotIterator.hasNext()) {
+      Snapshot remoteSnapshot = snapshotIterator.next();
+      assertNotNull(remoteSnapshot.id());
+      assertTrue(diskSet.contains(remoteSnapshot.snapshotId().snapshot()));
+      assertNotNull(remoteSnapshot.creationTimestamp());
+      assertNotNull(remoteSnapshot.status());
+      assertEquals(100L, (long) remoteSnapshot.diskSizeGb());
+      assertTrue(diskSet.contains(remoteSnapshot.sourceDisk().disk()));
+      assertNotNull(remoteSnapshot.sourceDiskId());
+      assertNotNull(remoteSnapshot.storageBytes());
+      assertNotNull(remoteSnapshot.storageBytesStatus());
+      count++;
+    }
+    assertEquals(2, count);
+    // test list snapshots with selected fields
+    snapshotPage = compute.listSnapshots(Compute.SnapshotListOption.filter(snapshotFilter),
+        Compute.SnapshotListOption.fields(Compute.SnapshotField.CREATION_TIMESTAMP));
+    snapshotIterator = snapshotPage.iterateAll();
+    count = 0;
+    while (snapshotIterator.hasNext()) {
+      Snapshot remoteSnapshot = snapshotIterator.next();
+      assertNull(remoteSnapshot.id());
+      assertTrue(diskSet.contains(remoteSnapshot.snapshotId().snapshot()));
+      assertNotNull(remoteSnapshot.creationTimestamp());
+      assertNull(remoteSnapshot.status());
+      assertNull(remoteSnapshot.diskSizeGb());
+      assertNull(remoteSnapshot.sourceDisk());
+      assertNull(remoteSnapshot.sourceDiskId());
+      assertNull(remoteSnapshot.storageBytes());
+      assertNull(remoteSnapshot.storageBytesStatus());
+      count++;
+    }
+    assertEquals(2, count);
+    compute.delete(firstDiskId);
```

Comment:
I generally don't wait when the method is not testing `delete` and there are no constraint on the maximum number of that resources that can be created. This allows the tests to run faster.